### ISSUE TITLE
lh-map-tools: Only jump to closing brack if inside pair

### ISSUE THREAD

**Issue by GoogleCodeExporter** [byzhang/lh-vim#3]
```
It should be possible to do the jump to ')' only
if we are within a pair of brackets -- all the pairs matching code is already
defined in various places.

This should be the case for ']' as well. 


```

Original issue reported on code.google.com by `ngaloppo@gmail.com` on 26 Feb 2008 at 8:41

**Comment by GoogleCodeExporter** [comment]
```
As it is not easy to know exactly whether we need to jump or add a closing 
bracket
character [1], I'll re-implement the mappings on closing-bracket keys with the
following heuristic: "if the next character matches the closing key, jump, 
otherwise
insert that character"

[1] in 
    f("foo", g(42^, 43)
If the cursor is on the '^', what should be the correct behaviour ? jump or  
insert ')' ?
```

Original comment by `luc.herm...@gmail.com` on 3 Mar 2008 at 4:06
- Changed title: **lh-map-tools: Only jump to closing brack if inside pair**
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
Feature implemented in the new kernel v1.0.0 of map-tools
```

Original comment by `luc.herm...@gmail.com` on 25 Mar 2008 at 2:24
- Changed state: **Fixed**
